src/components/PopupWithForm.js

```diff
@@ -30,4 +31,12 @@ export default class PopupWithForm extends Popup {
     this._popupForm.reset();
     super.close();
   }
+
+  setLoading(isLoading, saveButtonText) {
+    if (isLoading) {
+      this._saveButton.textContent = "Saving...";
+    } else {
+      this._saveButton.textContent = saveButtonText;
```

Comment:
Could be better: you can save the initial button text in the constructor and reuse it here


```js
// constructor
 this._saveButtonText = this._saveButton.textContent;

// here
 this._saveButton.textContent = this._saveButtonText;
```

Reply:
made change